Modules/SubModule_Import/xASL_imp_NII2BIDS_Session.m

```diff
@@ -0,0 +1,71 @@
+function xASL_imp_NII2BIDS_Session(imPar, bidsPar, studyPar, listSessions, nameSubjectSession, bidsLabel, iSession)
+%xASL_imp_NII2BIDS_SubjectSession NII2BIDS conversion for a single sessions.
+%
+% FORMAT: xASL_imp_NII2BIDS_SubjectSession(imPar, bidsPar, studyPar, listSessions, nameSubjectSession, subjectLabel, iSession)
+% 
+% INPUT:
+%   imPar                 - JSON file with structure with import parameter (STRUCT, REQUIRED)
+%   bidsPar               - Output of xASL_imp_Config (STRUCT, REQUIRED)
+%   studyPar              - JSON file with the BIDS parameters relevant for the whole study (STRUCT, REQUIRED)
+%   listSessions          - list of sessions (CELL ARRAY, REQUIRED, e.g.: {'ASL_1'})
+%   nameSubjectSession    - name of the subject (CELL ARRAY, REQUIRED)
+%   subjectLabel          - subject label (CHAR ARRAY, REQUIRED)
+%   iSession              - Session number (INTEGER, REQUIRED)
+%
+% OUTPUT:
+%  n/a
+%                         
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION: NII2BIDS conversion for a single sessions.
+% 
+% 1. Make a subject directory
+% 2. Iterate over runs
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% EXAMPLE:     n/a
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+    %% Unpack bidsLabel struct
+    subjectLabel = bidsLabel.subject;
+    visitLabel = bidsLabel.visit;
+
+    %% 1. Make a subject directory
+    if length(listSessions)>1 || ~isempty(visitLabel)
+        if ~isempty(visitLabel)
+            sessionLabel = ['ses-' visitLabel];
+        else
+            sessionLabel = ['ses-' listSessions{iSession}(5:end)];
+        end
+
+        if ~exist(fullfile(imPar.BidsRoot,['sub-' subjectLabel],sessionLabel),'dir')
```

Comment:
You can remove this if `~exist` line, and directly add the full folder with 'perf' at the end, `xASL_adm_CreateDir` always checks for each folder layer if it exists and create it.

Reply:
Sure, I changed it from mkdir to xASL_adm_CreateDir, so it kind of makes sense to remove the exist 9c36709 👍